ocaml/testsuite/tests/module-type-underscore/file_level.ml

```diff
@@ -0,0 +1,22 @@
+(* TEST
+   setup-ocamlc.byte-build-env;
+   compiler_reference = "${test_source_directory}/file_level.reference";
+   ocamlc_byte_exit_status = "2";
+   ocamlc.byte;
+   check-ocamlc.byte-output;
+  *)
+
+module type S = _
```

Comment:
Let's add some interesting tests playing with multiple files.

- `include List`, where `List` is a file-level module. Check if it's added to substitution properly.
- `module M = List` in both `.mli` and `.ml`. See if the substitution is added properly.
- `module type S = module type of List` in `.mli`, and then `module type S = _` .

Reply:
I've added a new test with `ocamlc_byte_exit_status = "0"` and an empty `.reference`, so at least we know we aren't getting type errors. Is this sufficient, or is there a nice way to print inferred type info (like in `toplevel` tests)?

Reply:
Pass `-i` to the compiler makes it print the inferred type of `.ml` without looking at `.mli`.